### PR TITLE
Add tenure_extend_timestamp to Block Response Accept messages

### PR DESCRIPTION
**Pull request by jferrant**
Needed to complete https://github.com/stacks-network/stacks-core/issues/5434

### CODE REVIEW

File: stacks-signer/src/tests/conf/signer-0.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 stacks_private_key = "6a1fc1a3183018c6d79a4e11e154d2bdad2d89ac8bc1b0a021de8b4d28774fbb01"
 node_host = "127.0.0.1:20443"
-endpoint = "localhost:30000"
```
Comment (jferrant) [inline stacks-signer/src/tests/conf/signer-0.toml:3]: Changed this to make config to str test pass.

**Review by hstove — CHANGES_REQUESTED** [review]
This needs to be backwards compatible - right now it'll fail if the payload doesn't have the timestamp. I think there's already a unit test with fixtures for backwards compatibility

**Comment by jferrant** [comment]
> This needs to be backwards compatible - right now it'll fail if the payload doesn't have the timestamp. I think there's already a unit test with fixtures for backwards compatibility

I mean I can easily make this default fill the timestamp with 0 if not found. Not sure that is considered backwards compatible enough. I don't actually see why this has to be backwards compatible though. It would of course require signers to upgrade, by why does this need to be backwards compatible?

**Comment by jferrant** [comment]
> > This needs to be backwards compatible - right now it'll fail if the payload doesn't have the timestamp. I think there's already a unit test with fixtures for backwards compatibility
> 
> I mean I can easily make this default fill the timestamp with 0 if not found. Not sure that is considered backwards compatible enough. I don't actually see why this has to be backwards compatible though. It would of course require signers to upgrade, by why does this need to be backwards compatible? I say this because we don't ever backprocess block responses.

**Comment by obycode** [comment]
> I don't actually see why this has to be backwards compatible though.

Since we are doing this without a hard fork, miners that upgrade should be able to talk to signers that have not upgraded, and vice versa.

**Comment by jferrant** [comment]
> > I don't actually see why this has to be backwards compatible though.
> 
> Since we are doing this without a hard fork, miners that upgrade should be able to talk to signers that have not upgraded, and vice versa.

haha I just answered this in my own response to Hank right before I saw this XD

**Review by hstove — APPROVED** [review]
LGTM! Should we merge this into a broader feature branch? This PR itself is probably fine to merge into develop, but with other PRs we'll probably want some other base branch.

**Comment by obycode** [comment]
> LGTM! Should we merge this into a broader feature branch? This PR itself is probably fine to merge into develop, but with other PRs we'll probably want some other base branch.

Yeah, that's a good point. Let's merge everything into `feat/time-based-tenure-extend` until we're ready.

**Review by obycode — APPROVED** [review]
LGTM

**Pull request merged**